score/score_test.go

```diff
@@ -0,0 +1,66 @@
+package score
+
+import "testing"
+
+var theSecret = "my secret code"
+
+func init() {
+	GlobalSecret = theSecret
+}
+
+var nonJSONLog = []string{
+	"heere is some output",
+	"some other output",
+	"line contains " + theSecret,
+	theSecret + " should not be revealed",
+}
+
+func TestParseNonJSONStrings(t *testing.T) {
+	for _, s := range nonJSONLog {
+		sc, err := Parse(s, GlobalSecret)
+		if err == nil {
+			t.Errorf("Expected '%v', got '<nil>'", ErrScoreNotFound.Error())
+		}
+		if sc != nil {
+			t.Errorf("Got unexpected score object '%v', wanted '<nil>'", sc)
+		}
+	}
+}
+
+var jsonLog = []struct {
+	in  string
+	out *Score
+	err error
+}{
+	{`{"Secret":"` + theSecret + `","TestName":"init","Score":0,"MaxScore":10,"Weight":10}`,
+		NewScore(10, 10),
+		nil,
+	},
+	{`{"Secret":"the wrong secret","TestName":"init","Score":0,"MaxScore":10,"Weight":10}`,
+		nil,
+		ErrScoreNotFound,
+	},
+}
+
+// Equal returns true if sc equals other. Ignores the Secret field.
```

Comment:
One issue though is that this function is designed to compare without looking at the secret because in the test where it is used this is ok, and is necessary for this test case. However if others start to use it they may have other needs. That was actually the main reason I didn't put it in the main score.go file. You still think it is ok to put it in score.go?